arch/arm/src/stm32wb/stm32wb_i2c.c

```diff
@@ -0,0 +1,2650 @@
+
+static inline void stm32wb_i2c_putreg32(struct stm32wb_i2c_priv_s *priv,
+                                        uint8_t offset, uint32_t value)
+{
+  putreg32(value, priv->config->base + offset);
+}
+
+/****************************************************************************
+ * Name: stm32wb_i2c_modifyreg32
+ *
+ * Description:
+ *   Modify a 32-bit register value by offset
+ *
+ ****************************************************************************/
+
+static inline
+void stm32wb_i2c_modifyreg32(struct stm32wb_i2c_priv_s *priv,
+                             uint8_t offset, uint32_t clearbits,
+                             uint32_t setbits)
+{
+  modifyreg32(priv->config->base + offset, clearbits, setbits);
+}
+
+/****************************************************************************
+ * Name: stm32wb_i2c_toticks
+ *
+ * Description:
+ *   Return a micro-second delay based on the number of bytes left to be
+ *   processed.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_STM32WB_I2C_DYNTIMEO
+static uint32_t stm32wb_i2c_toticks(int msgc, struct i2c_msg_s *msgs)
+{
+  size_t bytecount = 0;
+  int i;
+
+  /* Count the number of bytes left to process */
+
+  for (i = 0; i < msgc; i++)
+    {
+      bytecount += msgs[i].length;
+    }
+
+  /* Then return a number of microseconds based on a user provided scaling
+   * factor.
+   */
+
+  return USEC2TICK(CONFIG_STM32WB_I2C_DYNTIMEO_USECPERBYTE * bytecount);
+}
+#endif
+
+/****************************************************************************
+ * Name: stm32wb_i2c_enableinterrupts
+ *
+ * Description:
+ *   Enable I2C interrupts
+ *
+ ****************************************************************************/
+
+#ifndef CONFIG_I2C_POLLED
+static inline
+void stm32wb_i2c_enableinterrupts(struct stm32wb_i2c_priv_s *priv)
+{
+    stm32wb_i2c_modifyreg32(priv, STM32WB_I2C_CR1_OFFSET, 0,
+                            (I2C_CR1_TXRX | I2C_CR1_NACKIE));
+}
+#endif
+
+/****************************************************************************
+ * Name: stm32wb_i2c_sem_waitdone
+ *
+ * Description:
+ *   Wait for a transfer to complete
+ *
+ * There are two versions of this function.  The first is included when using
+ * interrupts while the second is used if polling (CONFIG_I2C_POLLED=y).
+ *
+ ****************************************************************************/
+
+#ifndef CONFIG_I2C_POLLED
+static inline
+int stm32wb_i2c_sem_waitdone(struct stm32wb_i2c_priv_s *priv)
+{
+  irqstate_t flags;
+  int ret;
+
+  flags = enter_critical_section();
+
+  /* Enable I2C interrupts */
+
+  /* The TXIE and RXIE interrupts are enabled initially in
+   * stm32wb_i2c_process. The remainder of the interrupts, including
+   * error-related, are enabled here.
+   */
+
+  stm32wb_i2c_modifyreg32(priv, STM32WB_I2C_CR1_OFFSET, 0,
+                        (I2C_CR1_ALLINTS & ~I2C_CR1_TXRX));
```

Comment:
Move the "(" to be below "p" of priv, unfortunately nxstyle doesn't get these alignment issues